daemon/list.go

```diff
@@ -61,6 +64,21 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			}
 		}
 	}
+
+	if ancestors, ok := psFilters["ancestor"]; ok {
+		ancestorFilter = true
+		byParents := daemon.Graph().ByParent()
+		// The idea is to walk the graph down the most "efficient" way.
+		for _, ancestor := range ancestors {
+			// First, get the imageId of the ancestor filter (yay)
+			image, err := daemon.Repositories().LookupImage(ancestor)
+			if err == nil {
+				// Then walk down the graph and put the imageIds in imagesFilter
+				imagesFilter = append(imagesFilter, populateImageFilterByParents(image.ID, byParents)...)
```

Comment:
If one of the ancestors happens to be a child of a previous 'ancestor' then we'll end up doing this recursive stuff twice for the same image ID (I think).  If so, we may need to add an if-statement before this line to check to see if 'ancestor' is already in the 'imagesFilter' list.  Just to optimize things a bit. But please double-check my thinking.